src/Microsoft.Extensions.ApiDescription.Client/DownloadFile.cs

```diff
@@ -94,19 +96,116 @@ public async Task<bool> ExecuteAsync()
                 return;
             }
 
-            log.LogMessage($"Downloading '{uri}' to '{destinationPath}'.");
+            log.LogMessage(MessageImportance.High, $"Downloading '{uri}' to '{destinationPath}'.");
 
-            using (var httpClient = new HttpClient
+            using (var httpClient = new HttpClient())
             {
-            })
+                await DownloadAsync(uri, destinationPath, httpClient, cancellationToken, log, timeoutSeconds);
+            }
+        }
+
+        public static async Task DownloadAsync(
+            string uri,
+            string destinationPath,
+            HttpClient httpClient,
+            CancellationToken cancellationToken,
+            TaskLoggingHelper log,
+            int timeoutSeconds)
+        {
+            // Timeout if the response has not begun within 1 minute
+            httpClient.Timeout = TimeSpan.FromMinutes(1);
+
+            var destinationExists = File.Exists(destinationPath);
+            var reachedCopy = false;
+            try
+            {
+                using (var response = await httpClient.GetAsync(uri, cancellationToken))
+                {
+                    response.EnsureSuccessStatusCode();
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    using (var responseStreamTask = response.Content.ReadAsStreamAsync())
```

Comment:
This would dispose the `Task` and not the content which is a bit unusual. Should we be disposing the latter instead?